sources/common/test/unitTest/stub/DummySignal.h

```diff
@@ -40,23 +40,20 @@ class DummySignal : public SIGNAL::Signal
         /// Dummy values used to construct dummy signal.
         /// @{
         static constexpr const char *DUMMY_NAME = "signal";
-        static const size_t DUMMY_SIZE = 32;
-        static const uint64_t DUMMY_TIMESTAMP = 0;
+        static constexpr size_t DUMMY_SIZE = 32;
+        static constexpr uint64_t DUMMY_TIMESTAMP = 0;
         static constexpr const char *DUMMY_VALUE = "0";
         static constexpr const char *DUMMY_TYPE = "dummy";
-        static const SIGNAL::SourceRegistry::HandleT DUMMY_HANDLE = 1;
+        static constexpr SIGNAL::SourceRegistry::HandleT DUMMY_HANDLE = 1;
         /// @}
 
         /// Dummy signal signal constructor.
         DummySignal(const std::string &name,
                     uint64_t timestamp,
                     const std::string &value = DUMMY_VALUE,
                     SIGNAL::SourceRegistry::HandleT source = DUMMY_HANDLE) :
-            Signal(name,
-                   DUMMY_SIZE,
-                   timestamp,
-                   DUMMY_TYPE,
-                   source),
+            Signal(std::make_unique<const SIGNAL::SignalDescriptor>(name, DUMMY_SIZE, DUMMY_TYPE, source).get(),
```

Comment:
Again, `unique_ptr` destructor will delete object almost immediately.

Reply:
OK. I went back to it. Initiall I thought it could be simplified but it turns out that still something similar would be needed. Ant the current implementation seems most simplistic. Let me explain.

Reply:
Ok, but code above produces a dangling pointer. `std::make_unique` results in a pointer that is destroyed immediately. You still have that pointer (you got it using `get()`) but that pointer is invalid, it points to nothing. Have a look: https://godbolt.org/z/jKjnnq - `new` then `delete` then dereference and assignment (`mov`). This is just dangling pointer.

Reply:
It applies to all cases when your `unique_ptr` or `vector` scope is shorter than scope of that pointer that you obtain using `get`. After that `vector` is destroyed that pointer you got is invalid.

Reply:
Oh yes, thanks for pointing this out. There was a static vector in one place and I haven't noticed this nasty bug there in the Dummy signal. Good catch :)

Reply:
`shared_ptr` won't change anything. Look: https://wandbox.org/permlink/MM1Xc8ywXn9doZsX "delete", then "dereference". You can allocate something and then discard object that own it.